extensions/amp-bind/0.1/amp-state.js

```diff
@@ -135,22 +150,39 @@ export class AmpState extends AMP.BaseElement {
       (getSourceOrigin(src) == getSourceOrigin(ampdoc.win.location))) {
       policy = UrlReplacementPolicy.ALL;
     }
-    return batchFetchJsonFor(
-        ampdoc, element, /* opt_expr */ undefined, policy, opt_refresh);
+
+    // If cross-origin="amp-viewer-auth-token-via-post" attribute is present,
+    // the viewer will make a remote xhr POST request with an auth token in the
+    // request body. Requires amp-viewer-assistance extension for auth token.
+    const crossOriginAttr = element.getAttribute('cross-origin');
+    if (crossOriginAttr &&
+        crossOriginAttr.indexOf('amp-viewer-auth-token-via-post') >= 0) {
+      return Services.viewerAssistanceForDocOrNull(ampdoc.win)
+          .then(viewerAssistance => {
+            devAssert(viewerAssistance,
+                'Viewer Assistance service can not be found');
```

Comment:
This can be a user error if the extension is not present. 

```js
userAssert(viewerAssistance, 'cross-origin="amp-viewer-auth-token-post" requires amp-viewer-assistance extension.');
```

Reply:
Done.